executor/README.md

```diff
@@ -0,0 +1,270 @@
+
+### Scheduler HTTP API
+
+#### `POST /rawscheduler`
+
+The job creation endpoint can be used to specify before and after commands, with the following keys:
+
+| key | type | description |
+|-----|------|-------------|
+| `before_commands` | array | An array of `command`s to run before all user specified commands. |
+| `after_commands` | array | An array of `command`s to run after all user specified commands. |
+
+Commands have the following structure:
+
+| key | type | description |
+|-----|------|-------------|
+| `value` | string | A string representing the command to be run, ex: `echo hello`. |
+| `guard` | boolean | An optional flag. If set, and the command returns a non-zero exit code, no further commands for the job will be executed. |
+| `async` | boolean | An optional flag. If set, the command runs asynchronously, in parallel with other future commands. It will be killed when all synchronous commands for a job are finished. |
+
+#### `GET /rawscheduler?job=:job`
+
+The job status endpoint can be used to view a job's progress, custom failure message, as well as the exit codes for all commands, by examining the following additional keys:
+
+| key | type    | description                              |
+| -------- | ------- | ---------------------------------------- |
+| `exit_code`     | integer | Status code when job specifies `command`  (will be absent if job has yet to finish). |
+| `before_exit_codes` | array of integers | Status codes for `before_commands`. |
+| `after_exit_codes` | array of integers | Status codes for `after_commands`. |
+| `progress` | long | Progress reported by a job's commands via the executor HTTP API. May be absent. |
+| `message` | string | A custom failure message set by a command using the executor HTTP API. May be absent. |
+
+### Executor HTTP API
+
+The executor provides an HTTP API that can be used from a job's commands. It currently only supports two endpoint:
+
+#### `PATCH /task/:taskid`
+
+This request also requires a body. Since it is a `PATCH` request, a valid request may contain only some of the following keys:
+
+| key | type   | description                              |
+| -------- | ------ | ---------------------------------------- |
+| `progress` | float  | A number between 0 and 100, representing the progress of the task |
+| `message`   | string | A custom failure reason, expected to be accompanied by a non-zero exit code. |
+| `env`      | object | A map of env variable names to values, a null value will unset the variable. This is really only useful from a before command, since it will only affect commands that run afterward.  |
```

Comment:
Are there restrictions on the length of the names or values of the environment variables?